File: SensorServiceDemo/app/src/main/java/com/example/sensorservicedemo/MainActivity.java
package com.example.sensorservicedemo;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    SensorReceiver receiver;
    Intent intent; // 센서 서비스를 구동하는 데 사용하는 인텐트
    public TextView tv_accX, tv_accY, tv_accZ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tv_accX = findViewById(R.id.tv_accX);
        tv_accY = findViewById(R.id.tv_accY);
        tv_accZ = findViewById(R.id.tv_accZ);

        intent = new Intent(getApplicationContext(), SensorService.class);
        startService(intent);   // 서비스를 시작한다. - SensorService객체의 onStartCommand() 호출
   }

    @Override
    protected void onStop() {
        super.onStop();
        if (receiver != null){ // 방송 수신자 등록 해제
            unregisterReceiver(receiver);
            receiver = null;
        }
        stopService(intent);   // 센서서비스 중지
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (receiver != null){ // 방송 수신자 등록 해제
            unregisterReceiver(receiver);
            receiver = null;
        }
        stopService(intent);   // 센서서비스 중지
    }

    @Override
    protected void onResume() {
        super.onResume();
        receiver = new SensorReceiver(MainActivity.this);
        IntentFilter filter = new IntentFilter();
        filter.addAction(SensorService.MY_ACTION_SENSOR);
        registerReceiver(receiver, filter);
        startService(intent);
    }
}

File: MusicPlayerDemo/app/src/main/java/com/example/musicplayerdemo/MainActivity.kt
package com.example.musicplayerdemo

import android.content.ComponentName
import android.content.ServiceConnection
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder
import android.widget.Button

class MainActivity : AppCompatActivity() {

    lateinit var btnPlay: Button
    lateinit var btnPause: Button
    lateinit var btnStop: Button

    var mService: MusicPlayerService? = null;   // 서비스 변수

    // 서비스와 구성요소 연결 상태 모니터링
    val mServiceConnection = object : ServiceConnection{
        // 만약 서비스가 연결되면...
        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
            // service 참조변수를  MusicPlayerService의 MusicPlayerBinder객체로 변환
            // 변환된 Binder객체의 getService()함수로 서비스 객체를 받음
            mService = (service as MusicPlayerService.MusicPlayerBinder).getService()
        }

        override fun onServiceDisconnected(name: ComponentName?) {
            mService = null // 서비스가 끊기면 mService를 null로 만든다.
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnPlay = findViewById(R.id.btn_play)
        btnPause = findViewById(R.id.btn_pause)
        btnStop = findViewById(R.id.btn_stop)

        btnPlay.setOnClickListener{
            play()
        }

        btnPause.setOnClickListener{
            puase()
        }

        btnStop.setOnClickListener{
            stop()
        }
    }

    private fun stop() {
        TODO("Not yet implemented")
    }

    private fun puase() {
        TODO("Not yet implemented")
    }

    private fun play() {
        TODO("Not yet implemented")
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }
}

